model/src/main/scala/uk/gov/ons/addressIndex/model/server/response/AddressResponse.scala

```diff
@@ -189,9 +198,22 @@ object AddressResponseAddress {
       northing <- Try(other.northing.toInt)
     } yield AddressResponseGeo(latitude, longitude, easting, northing)
 
+    val saoHyphen = if ((other.saoStartNumber != "" || other.saoStartSuffix != "")
+      && (other.saoEndNumber != "" || other.saoEndSuffix != "")) "-" else ""
+    val saoNumbers = Seq(other.saoStartNumber, other.saoStartSuffix, saoHyphen, other.saoEndNumber, other.saoEndSuffix).mkString("")
+    val sao = if (other.saoText == other.organisation) saoNumbers else s"$saoNumbers ${other.saoText}"
+
+    val paoHyphen = if ((other.paoStartNumber != "" || other.paoStartSuffix != "")
+      && (other.paoEndNumber != "" || other.paoEndSuffix != "")) "-" else ""
+    val paoNumbers = Seq(other.paoStartNumber, other.paoStartSuffix, paoHyphen, other.paoEndNumber, other.paoEndSuffix).mkString("")
```

Comment:
`s"${var.x} ${var.y} ${var.q}".split(" ").mkString(" ")` is harder to read and understand (for me) than
`Seq(var.x, var.y, var.q).filter(_.nonEmpty).mkString(" ")`
In first case I needed to spend some time to understand what actually happens (how the double spaces are removed), in the second case it is more explicit.